hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -1809,6 +1809,37 @@ def accept(*args, **kwargs):
     return accept
 
 
+@defines_strategy
+@composite
+def subsequence_of(
+    draw,  # type: ignore
+    elements,  # type: SearchStrategy[Ex]
+    min_size=None,  # type: int
+    average_size=None,  # type: int
+    max_size=None,  # type: int
+):
+    # type: (...) -> List[Ex]
+    """Returns a strategy which generates sub-sequences of the elements
+    sequence.
+    """
+    check_valid_sizes(min_size, average_size, max_size)
+
+    # TODO: fix hax to always get numerical constraints.
+    min_size = 0 if min_size is None else min_size
+    max_size = len(elements) if max_size is None else max_size
+
+    def element_mask():
+        # type: () -> List[bool]
+        always_size = min_size
+        never_size = len(elements) - max_size
+        maybe_size = len(elements) - always_size - never_size
+        choices = ([True] * always_size + [draw(booleans()) for _ in range(maybe_size)] + [False] * never_size)
```

Comment:
I wonder if it's better for this to be `[True] * draw(integers(min_size, max_size))`, and then padded with `False` up to `len(elements)`.

If the plan is to run the mask through `permutations`, then I don't think it's necessary to flip individual coins for each of the middle bits in the mask.

This should also give a more even distribution of subsequence lengths, instead of clustering near 50%.